contrib/benchmarkDictBuilder/benchmark.c

```diff
@@ -0,0 +1,458 @@
+
+_cleanup:
+  freeDictInfo(dInfo);
+  return result;
+}
+
+
+
+/** benchmarkLegacy() :
+ *  Measure how long legacy dictionary builder takes and compression ratio with the legacy dictionary
+ *  @return 0 if benchmark successfully, 1 otherwise
+ */
+int benchmarkLegacy(sampleInfo *srcInfo, unsigned maxDictSize, ZDICT_legacy_params_t *legacyParam) {
+  const int displayLevel = legacyParam->zParams.notificationLevel;
+  int result = 0;
+  clock_t t;
+  t = clock();
+  dictInfo* dInfo = createDictFromFiles(srcInfo, maxDictSize, NULL, NULL, legacyParam);
+  t = clock() - t;
+  double time_taken = ((double)t)/CLOCKS_PER_SEC;
+  if (!dInfo) {
+    DISPLAYLEVEL(1, "LEGACY does not train successfully\n");
+    result = 1;
+    goto _cleanup;
+
+  }
+  DISPLAYLEVEL(2, "LEGACY took %f seconds to execute \n", time_taken);
+
+  double cRatio = compressWithDict(srcInfo, dInfo, legacyParam->zParams.compressionLevel, displayLevel);
+  if (cRatio < 0) {
+    DISPLAYLEVEL(1, "Compressing with LEGACY dictionary does not work\n");
+    result = 1;
+    goto _cleanup;
+
+  }
+  DISPLAYLEVEL(2, "Compression ratio with legacy dictionary is %f\n", cRatio);
+
+_cleanup:
+  freeDictInfo(dInfo);
+  return result;
+}
+
+
+
+int main(int argCount, const char* argv[])
+{
+  int displayLevel = 2;
+  const char* programName = argv[0];
+  int result = 0;
+  /* Initialize arguments to default values */
+  unsigned k = 200;
+  unsigned d = 6;
+  unsigned cLevel = 3;
+  unsigned dictID = 0;
+  unsigned maxDictSize = g_defaultMaxDictSize;
+
+  /* Initialize table to store input files */
+  const char** filenameTable = (const char**)malloc(argCount * sizeof(const char*));
+  unsigned filenameIdx = 0;
+
+  char* fileNamesBuf = NULL;
+  unsigned fileNamesNb = filenameIdx;
+  int followLinks = 0;
+  const char** extendedFileList = NULL;
+
+  /* Parse arguments */
+  for (int i = 1; i < argCount; i++) {
+    const char* argument = argv[i];
+    if (longCommandWArg(&argument, "in=")) {
+      filenameTable[filenameIdx] = argument;
+      filenameIdx++;
+      continue;
+    }
+    DISPLAYLEVEL(1, "benchmark: Incorrect parameters\n");
+    return 1;
+  }
+
+
+  /* Get the list of all files recursively (because followLinks==0)*/
+  extendedFileList = UTIL_createFileList(filenameTable, filenameIdx, &fileNamesBuf,
+                                        &fileNamesNb, followLinks);
+  if (extendedFileList) {
+    unsigned u;
+    for (u=0; u<fileNamesNb; u++) DISPLAYLEVEL(4, "%u %s\n", u, extendedFileList[u]);
+    free((void*)filenameTable);
+    filenameTable = extendedFileList;
+    filenameIdx = fileNamesNb;
+  }
+
+  size_t blockSize = 0;
+  sampleInfo* srcInfo= getSampleInfo(filenameTable,
+                    filenameIdx, blockSize, maxDictSize, displayLevel);
+
+  /* set up zParams */
+  ZDICT_params_t zParams;
+  zParams.compressionLevel = cLevel;
+  zParams.notificationLevel = displayLevel;
+  zParams.dictID = dictID;
+
+  /* for random */
+  ZDICT_random_params_t randomParam;
```

Comment:
nit: prefer to put these in their own scope. This way I can clearly see that these variables won't be used later on in the function, and you can't accidentally use the wrong variable, like when checking error codes.

```c
{
  ZDICT_random_params_t randomParams;
  // ...
  if (randomResult) { /* ... */ }
}
```